src/server/server.cc

```diff
@@ -910,7 +910,7 @@ void Server::GetReplicationInfo(std::string *info) {
     if (slave->IsStopped()) continue;
 
     string_stream << "slave" << std::to_string(idx) << ":";
-    string_stream << "ip=" << slave->GetConn()->GetIP() << ",port=" << slave->GetConn()->GetListeningPort()
+    string_stream << "ip=" << slave->GetConn()->GetAnnounceIPOrIP() << ",port=" << slave->GetConn()->GetListeningPort()
```

Comment:
Here seems we use `slave->GetConn()->GetListeningPort()` instead of the announce-port, is it right?

Reply:
GetListeningPort contains announce-port

see function `ReplicationThread::CBState ReplicationThread::replConfWriteCB(bufferevent *bev, void *ctx) {`